cardano-cli/src/Cardano/CLI/EraBased/Run/Key.hs

```diff
@@ -115,27 +131,27 @@ runNonExtendedKeyCmd
   writeVerificationKey ssk =
     case ssk of
       APaymentExtendedVerificationKey vk ->
-        writeToDisk vkf Nothing (castVerificationKey vk :: VerificationKey PaymentKey)
+        writeToDisk vkf paymentVkeyDesc (castVerificationKey vk :: VerificationKey PaymentKey)
       ADRepExtendedVerificationKey vk ->
-        writeToDisk vkf (Just drepKeyEnvelopeDescr) (castVerificationKey vk :: VerificationKey DRepKey)
+        writeToDisk vkf drepVkeyDesc (castVerificationKey vk :: VerificationKey DRepKey)
       AStakeExtendedVerificationKey vk ->
-        writeToDisk vkf Nothing (castVerificationKey vk :: VerificationKey StakeKey)
+        writeToDisk vkf stakeVkeyDesc (castVerificationKey vk :: VerificationKey StakeKey)
       AGenesisExtendedVerificationKey vk ->
-        writeToDisk vkf Nothing (castVerificationKey vk :: VerificationKey GenesisKey)
+        writeToDisk vkf genesisVkeyDesc (castVerificationKey vk :: VerificationKey GenesisKey)
       AGenesisDelegateExtendedVerificationKey vk ->
-        writeToDisk vkf Nothing (castVerificationKey vk :: VerificationKey GenesisDelegateKey)
+        writeToDisk vkf genesisVkeyDelegateDesc (castVerificationKey vk :: VerificationKey GenesisDelegateKey)
       nonExtendedKey -> left $ KeyCmdExpectedExtendedVerificationKey nonExtendedKey
 
 
   writeToDisk
    :: Key keyrole
    => File content Out
-   -> Maybe TextEnvelopeDescr
+   -> TextEnvelopeDescr
```

Comment:
I wouldn't force the caller to provide a description because it's not necessary.

Reply:
But don't we want to entice the future new callers to provide a description :upside_down_face:?

Reply:
@Jimbo4350> I reverted to using the `Maybe` and so I'm resolving this thread :+1: